src/components/composer/composer/composer.ts

```diff
@@ -335,71 +324,38 @@ export class Composer extends Component<ComposerProps, SpreadsheetChildEnv> {
     } else {
       ev.stopPropagation();
     }
-    const { start, end } = this.contentHelper.getCurrentSelection();
-    if (
-      !this.env.model.getters.isSelectingForComposer() &&
-      !(ev.key === "Enter" && (ev.altKey || ev.ctrlKey))
-    ) {
-      this.env.model.dispatch("CHANGE_COMPOSER_CURSOR_SELECTION", { start, end });
-      this.isKeyStillDown = true;
-    }
   }
 
   /*
    * Triggered automatically by the content-editable between the keydown and key up
    * */
-  onInput() {
+  onInput(ev: InputEvent) {
     if (this.props.focus === "inactive" || !this.shouldProcessInputEvents) {
       return;
     }
+    let content = this.contentHelper.getText();
+    let selection = this.contentHelper.getCurrentSelection();
+    if (ev.inputType === "insertParagraph") {
+      content = content.slice(0, selection.start) + NEWLINE + content.slice(selection.end);
```

Comment:
`selection.start` can be >  `selection.end`n and it makes this bug. It's probably something we should handle in `contentHelper.getCurrentSelection()`